app/experimenter/base/tests/test_base.py

```diff
@@ -49,3 +52,39 @@ def test_app_version_file_error(self):
                 version = app_version()
                 mf.assert_called_once_with(settings.APP_VERSION_JSON_PATH)
                 self.assertEqual(version, "")
+
+
+class TestGetUploadsStorage(TestCase):
+    def setUp(self):
+        get_uploads_storage.cache_clear()
+
+    @override_settings(UPLOADS_GS_CREDENTIALS=None)
+    def test_get_uploads_storage_file(self):
+        storage = get_uploads_storage()
+        self.assertIsInstance(storage, FileSystemStorage)
+
+    @mock.patch("experimenter.base.service_account")
+    @mock.patch("experimenter.base.GoogleCloudStorage")
+    def test_get_uploads_storage_gcp(self, MockGoogleCloudStorage, mock_service_account):
+        expected_bucket_name = "bazquux"
+        expected_credentials = "/app/uploads-credentials.json"
+
+        with self.settings(
+            UPLOADS_GS_BUCKET_NAME=expected_bucket_name,
+            UPLOADS_GS_CREDENTIALS=expected_credentials,
+        ):
```

Comment:
you can also do this as decorators too using [override_settings](https://docs.djangoproject.com/en/3.2/topics/testing/tools/#django.test.override_settings)

Reply:
Not as easy to use as a decorator when I want to use those variables later in assertions

Reply:
Yep not a big deal, if they're just hard coded you can hard code them in both places, or make them class vars, or whatever